zebra-chain/src/transaction/serialize.rs

```diff
@@ -335,7 +364,67 @@ impl ZcashDeserialize for Transaction {
                 let inputs = Vec::zcash_deserialize(&mut reader)?;
                 let outputs = Vec::zcash_deserialize(&mut reader)?;
 
-                // TODO: deserialize sapling shielded data according to the V5 transaction spec
+                let value_balance =
+                    amount::Amount::<amount::NegativeAllowed>::zcash_deserialize(&mut reader)?;
```

Comment:
Rust can do type inference and conversion here:
```suggestion
                let value_balance = (&mut reader).zcash_deserialize_into()?;
```